### ISSUE TITLE
Property 'options' does not exist on <SelectField /> (uniforms-semantic)

### ISSUE THREAD

**Issue by kestarumper** [vazco/uniforms#806]
```
Type '{ name: string; options: { label: string; value: string; }[]; onChange: (repositoryId: string) => void; }' is not assignable to type 'IntrinsicAttributes & { label?: ReactNode; name: string; placeholder?: string | boolean | null | undefined; } & Pick<Partial<GuaranteedProps<string | string[] | undefined>>, "id" | ... 9 more ... | "showInlineError"> & Pick<...> & { ...; }'.
  Property 'options' does not exist on type 'IntrinsicAttributes & { label?: ReactNode; name: string; placeholder?: string | boolean | null | undefined; } & Pick<Partial<GuaranteedProps<string | string[] | undefined>>, "id" | ... 9 more ... | "showInlineError"> & Pick<...> & { ...; }'.
```
**Versions**:
- uniforms `3.0.0-rc3`
- uniforms-semantic `3.0.0-rc3`

**Description**:
Property `options` do not exist but are used to calculate `allowedValues`. The only problem is that they're missing in the typescript definitions.

**Comment by radekmie** [comment]
As this topic arose once again (internally), I thought it'd be great to describe the problem _(I'm sure I already did that but there's no comment here)_.

All props received by a field created with `connectField` (or directly using `useField`) are extended with the ones from a bridge method called `getProps`. All of the standard bridges understand the `options` prop (e.g., [`JSONSchemaBridge#getProps`](https://github.com/vazco/uniforms/blob/be843c718eec029211d08448cb8159b96be81449/packages/uniforms-bridge-json-schema/src/JSONSchemaBridge.ts#L270-L280) code), but a new or custom bridge may not. Technically speaking, we'd need to have a way, to extend props (their types) from the bridge.

**Comment by radekmie** [comment]
We'll leave it for now, as there's no easy way of handling it.

**Comment by radekmie** [comment]
Instead of using `allowedValues` as defined in #862, we'll use `options` prop instead. And it won't be resolved on the bridge level, but on the field instead.

**Comment by kestarumper** [comment]
We've discussed this, and there will be changes on the bridge level, as @radekmie [mentioned above](https://github.com/vazco/uniforms/issues/806#issuecomment-990865222).

## Breaking changes
We'll be removing `allowedValues` and `transform` properties. Fields will only use `options` property.

## The common API
Bridges will use
```ts
type Option = {
    disabled?: boolean;
    key?: string
    label?: string;
    value: unknown;
}
```
(similar as described in [#862](https://github.com/vazco/uniforms/issues/862#issuecomment-990863975))

## Fields
Fields should also follow this format but can also extend it for custom use cases.
For example, the field can assume that the `value: unknown` is of type `value: string` in SemanticUI but will be different `value: string | number` in AntD. One might also extend the type and add other properties as well (field level).

## Migrating
- Usage of `allowedValues` and `transform` should be replaced with `options`
- `SimpleSchema` allowed you to provide a function for the `options` property - you will be able to retain this by converting that function to a getter using `get` keyword.

**Comment by kestarumper** [comment]
Closes #1120
